Add Trino 456 release notes (#23183)
trinodb/trino: docs/src/main/sphinx/release/release-456.md

# Release 456 (4 Sep 2024)

## General

* Expose additional JMX metrics about resource groups, including CPU and memory
  usage, limits, and scheduling policy. ({issue}`22957`)
* Add [adaptive plan optimization](/optimizer/adaptive-plan-optimizations) which
  can reorder joins at runtime. The feature can be disabled with the
  `fault-tolerant-execution-adaptive-join-reordering-enabled` configuration
  property or the `fault_tolerant_execution_adaptive_join_reordering_enabled`
  session property. ({issue}`23046`)

## JDBC driver

> * Add publishing a JDBC driver JAR without shaded, third-party dependencies. ({issue}`22098`)

Reviewer: “Publish a jdbc driver…”. Also, replace “shaded” with “bundled”

## BigQuery connector

> * Fix failures with queries using table functions when `parent-project-id` is

Author: @pajaks and @ebyhr is this okay? I dont think we can use PTF in the release notes and I am not sure if this is now correct. Did you mean only the query pass through table function?

Reviewer 2: Looks good to me

  defined. ({issue}`23041`)

## Blackhole connector

* Add support for replacing tables. ({issue}`23004`)

## Delta Lake connector

> * Add support for creating tables with deletion vectors. ({issue}`22104`)

Reviewer: Is the concept of “deletion vector” visible to users? If it’s a feature in Delta Lake that users have to be aware of, add a link to their docs.

Author: Linking like we do in older release notes now.. maybe we need to add a docs section at some stage @findinpath @jhlodin

> * Add support for lock-less Delta Lake write reconciliation on S3. ({issue}`23145`)

Reviewer: What’s the user visible impact of this? “Lockless” and “write reconciliation” are implementation-level concepts.

Author: Reworded and pushed

Reviewer: I still wonder whether we need to mention lockless and write reconciliation. Are those well-known terms in the Delta Lake spec or just details of our implementation?

# Release 456 (5 Sep 2024)

## General

* Expose additional JMX metrics about resource groups, including CPU and memory
  usage, limits, and scheduling policy. ({issue}`22957`)

> * Add [adaptive plan optimization](/optimizer/adaptive-plan-optimizations) which
>   can reorder joins at runtime. The feature can be disabled with the

Reviewer 3: ```suggestion
* Add [adaptive plan optimization](/optimizer/adaptive-plan-optimizations) which
  can reorder joins at runtime when FTE is used. The feature can be disabled with the
```

Only works with FTE it seems. The name of the property implies that but good to be explicit.

Author: The docs have that information already and are linked right above.. I just wanted to keep it shorter. Wdyt @martint ?

Reviewer: I would also rephrase it in terms of user-visible impact:

> Improve performance of queries involving joins when fault tolerant execution is enabled. This optimization can be disabled via the xxx configuration ...

(eventually, we're going to make this optimizations always-on and remove the fallback)

  `fault-tolerant-execution-adaptive-join-reordering-enabled` configuration
  property or the `fault_tolerant_execution_adaptive_join_reordering_enabled`
  session property. ({issue}`23046`)
* Improve performance for LZ4, zstd, and Snappy compression and decompression of
  ORC and Parquet files used for [exchange spooling with fault-tolerant
  execution mode](fte-exchange-manager). ({issue}`22532`)
* Extend the Kafka event listener to send split completion events. ({issue}`23065`)

## JDBC driver

> * Publishing a JDBC driver JAR without bundled, third-party dependencies. ({issue}`22098`)

Author: Fixed

> * Publishing a JDBC driver JAR without bundled, third-party dependencies. ({issue}`22098`)

Reviewer 3: ```suggestion
* A new JDBC driver JAR without bundled, third-party dependencies is now available as <original-jar-name>-non-shaded. ({issue}`22098`)
```

(please fill in original jar name, I don't remember what it's called on Maven central).

Author: We dont need that level of detail in the release notes. I already updated the docs and will just link to it here.

## BigQuery connector

* Fix failures with queries using table functions when `parent-project-id` is
  defined. ({issue}`23041`)

## Blackhole connector

> * Add support for replacing tables. ({issue}`23004`)

Reviewer 3: Might be better to say `CREATE OR REPLACE TABLE` support.

Author: Update to same wording we used in Delta Lake in the past

## Delta Lake connector

* Add support for creating tables with
  [deletion vector](https://docs.delta.io/latest/delta-deletion-vectors.html).
  ({issue}`22104`)
* Improve performance for concurrent write operations on S3 by using lock-less
  Delta Lake write reconciliation. ({issue}`23145`)
* Improve performance for [file compression and decompression](file-compression)

>   of Parquet files. ({issue}`22532`)

Reviewer 3: is only parquet true?

Author: Yeah .. that was the thinking .. but you are right .. maybe we should omit file formats .. wdyt @martint @dain and @wendigo ?

* Fix SSE configuration when using S3SecurityMapping with kmsKeyId configured. ({issue}`23299`)

## Hive connector

* Improve performance of queries that scan a large number of partitions. ({issue}`23194`)
* Improve performance for [file compression and decompression](file-compression)

>   of ORC and Parquet files. ({issue}`22532`)

Reviewer 3: only ORC and PArquet? What about JSON, CSV etc.

Reviewer 4: cc @wendigo

Author: See other comment .. maybe we remove the file format mentions

Reviewer: It's only for LZ4, Snappy and ZSTD, for any format that supports those compression schemes.

## Hudi connector

* Improve performance for [file compression and decompression](file-compression)
  of Parquet files. ({issue}`22532`)

## Iceberg connector

* Improve performance for [file compression and decompression](file-compression)
  of ORC and Parquet files. ({issue}`22532`)

## Memory connector

> * Add support for renaming schemas. ({issue}`22659`)

Reviewer 3: In past for SQL support changes we used to say like `Implement `ALTER SCHEMA ... RENAME` or something like that.

Author: Updated

## Prometheus connector

* Fix reading large Prometheus responses. ({issue}`23025`)

## SPI

* Remove the deprecated `ConnectorMetadata.createView` method. ({issue}`23208`)
* Remove the deprecated `ConnectorMetadata.beginRefreshMaterializedView` method
  without the `RefreshType` argument. ({issue}`23212`)
* Remove the deprecated `ConnectorMetadata.finishInsert` method without the
  `sourceTableHandles` argument. ({issue}`23213`)
* Remove the deprecated `ConnectorMetadata.createTable(ConnectorSession session,
  ConnectorTableMetadata tableMetadata, boolean ignoreExisting)` method.
  ({issue}`23209`)
* Remove the deprecated `ConnectorMetadata.createTable(ConnectorSession session,
  ConnectorTableMetadata tableMetadata, boolean ignoreExisting)` method.
  ({issue}`23209`)
* Remove the deprecated `ConnectorMetadata.beginCreateTable` method without the
  `replace` argument. ({issue}`23211`)

> * Remove the deprecated `ConnectorSplit.getInfo method` method. Use the

Reviewer 3: ```suggestion
* Remove the deprecated `ConnectorSplit.getInfo` method. Use the
```

  `getSplitInfo` method instead. ({issue}`23271`)
* Remove the deprecated `DecimalConversions.realToShortDecimal` method. Use the
  `realToShortDecimal(float value, long precision, long scale)` method instead.

>   ({issue}`23275`)

Reviewer 3: this reads weirdly since we say replace method with same method name with no idea how the old method's signature looked like.

Author: Fixed.
